tests/integration/object/user/userkeys/userkeys.go

```diff
@@ -0,0 +1,837 @@
+}
+
+func checkStatusKeys(t *testing.T, k8sh *utils.K8sHelper, osu cephv1.CephObjectStoreUser, expectedSecrets []*corev1.Secret) {
+	t.Run(fmt.Sprintf("cephObjectStoreUser %q has .status.keys set", osu.Name), func(t *testing.T) {
+		ctx := context.TODO()
+
+		liveOsu, err := k8sh.RookClientset.CephV1().CephObjectStoreUsers(ns.Name).Get(ctx, osu.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+
+		require.NotNil(t, liveOsu.Status)
+		assert.Len(t, liveOsu.Status.Keys, len(expectedSecrets))
+
+		for _, secret := range expectedSecrets {
+			secretRef, err := func(secretName string, keys []cephv1.SecretReference) (cephv1.SecretReference, error) {
+				for _, secretRef := range keys {
+					if secretRef.Name == secretName {
+						return secretRef, nil
+					}
+				}
+				return cephv1.SecretReference{}, fmt.Errorf("secretReference for secret %q not found in CephObjectStoreUser.status.keys", secret.Name)
+			}(secret.Name, liveOsu.Status.Keys)
+			require.NoError(t, err)
+
+			// fetch the live secret for UID and ResourceVersion
+			liveSecret, err := k8sh.Clientset.CoreV1().Secrets(ns.Name).Get(ctx, secret.Name, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			assert.Equal(t, liveSecret.Name, secretRef.Name)
+			assert.Equal(t, liveSecret.Namespace, secretRef.Namespace)
+			assert.Equal(t, liveSecret.UID, secretRef.UID)
+			assert.Equal(t, liveSecret.ResourceVersion, secretRef.ResourceVersion)
+		}
+	})
+}
+
+func checkRgwUserKeys(t *testing.T, adminClient *admin.API, osu cephv1.CephObjectStoreUser, expectedSecrets []*corev1.Secret) {
+	t.Run(fmt.Sprintf("rgw user %q has keys set", osu.Name), func(t *testing.T) {
+		ctx := context.TODO()
+
+		for _, secret := range expectedSecrets {
+			var keySpec admin.UserKeySpec
+
+			// assume that the .Phase doesn't change when updating keys
+			inSync := utils.Retry(40, time.Second, fmt.Sprintf("CephObjectStoreUser has key in sync with secret %q ", secret.Name), func() bool {
+				liveUser, err := adminClient.GetUser(ctx, admin.User{ID: osu.Name})
+				if err != nil {
+					return false
+				}
+				keySpec, err = findUserKeySpec(liveUser.Keys, string(secret.Data["AWS_ACCESS_KEY_ID"]))
+				return err == nil
+			})
+			require.True(t, inSync)
+
+			assert.Equal(t, string(secret.Data["AWS_ACCESS_KEY_ID"]), keySpec.AccessKey)
+			assert.Equal(t, string(secret.Data["AWS_SECRET_ACCESS_KEY"]), keySpec.SecretKey)
+		}
+
+		// check that no extra keys are present
+		liveUser, err := adminClient.GetUser(ctx, admin.User{ID: osu.Name})
+		require.NoError(t, err)
+		assert.Equal(t, len(expectedSecrets), len(liveUser.Keys))
+	})
+}
+
+func TestObjectStoreUserKeys(t *testing.T, k8sh *utils.K8sHelper, installer *installer.CephInstaller, logger *capnslog.PackageLogger, tlsEnable bool) {
+	t.Run("ObjectStoreUser keys", func(t *testing.T) {
+		if tlsEnable {
+			// Skip testing with and without TLS to reduce test time
+			t.Skip("skipping test for TLS enabled clusters")
+		}
```

Comment:
I think this is a good call. Thanks!

Reply:
I feel that the current approach of running all of the tests twice (except the same skip I added to the bucket owner tests), with and without tls configured, isn't a good approach and it doubles the runtime without any real benefit.  I'm causally thinking about how to restructure the object suite to make it easier to maintain and to reduce runtime.  My current thoughts are that the tls configuration should be standalone with a few simple checks to make sure the api is working as expected and to run everything else without tls.  My assumption (which could be wrong) is that most production deployments are terminating tls with an ingress controller instead of at rgw.